bundles/org.openhab.binding.neohub/src/test/java/org/openhab/binding/neohub/test/NeoHubProtocolTests.java

```diff
@@ -70,7 +77,15 @@ void testWssConnection() throws NeoHubException, IOException {
             config.socketTimeout = SOCKET_TIMEOUT;
             config.apiToken = HUB_API_TOKEN;
 
-            NeoHubWebSocket socket = new NeoHubWebSocket(config, "test");
+            SslContextFactory.Client sslContextFactory = new SslContextFactory.Client();
```

Comment:
There are three problems here:
1. When RUN_WSS_TEST is true and HUB_IP_ADDRESS is a valid value, the JUnit test does not exit at line 70, but actually proceeds to line 85; however the test then freezes at that line.
2. Not only should the Junit test not freeze, but if the value of HUB_API_TOKEN is valid , then the test should make an actual live online test to an existing online NeoHub; however with a mocked web socket factory, this will probably fail.
3. The code probably also needs `assertNonNull(webSocketFactory)` to eliminate a compiler warning.

```
    private static final String HUB_IP_ADDRESS = "192.168.1.xxx"; <= VALID
    private static final String HUB_API_TOKEN = "12345678-1234-1234-1234-123456789ABC"; <= VALID
    private static final boolean RUN_WSS_TEST = false; <= TRUE
```

Reply:
I agree, these unit tests have no sense. They are even doing nothing with the current code because a constant is not set to true. I have the strange feeling they were implemented by the author for testing in real conditions and then disabled by setting constants.
I don't know what to do with that.
I do not plan to spend more time on that PR, so I can close it if this is easier.

Reply:
> I have the strange feeling they were implemented by the author for testing in real conditions and then disabled by setting constants.

Correct.

> I do not plan to spend more time on that PR, so I can close it if this is easier.

Your choice. The ideal solution would be to fix the JUnit code so that it can actually do the tests. That would mean creating a real webSocketFactory instead of using `mock`..